src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonProperty.cs

```diff
@@ -17,9 +17,9 @@ namespace System.Text.Json
         ///   The value of this property.
         /// </summary>
         public JsonElement Value { get; }
```

Comment:
The `JsonProperty.NameEquals` API that takes string can work with null text. Please update the ref accordingly.

`NameEquals(string text)` -> `NameEquals(string? text)`